_posts/2021/2021-01-25-memoize-expensive-code.md

```diff
@@ -0,0 +1,73 @@
+---
+title: "Memoize Expensive Code"
+description: "The ||= operator can provide significant performance improvements"
+layout: article
+category: ruby
+image:
+  base: '2021/memoize-expensive-code'
+  alt: "Red Floppy Disk"
+  credit:  Fredy Jacob
+  source: "https://unsplash.com/photos/t0SlmanfFcg"
+---
+
+Memoization is a performance optimization where the result of a slow or non-performant piece of code is temporarily stored and when the expensive code is called again, the stored value is returned.
+
+It is a specific form of caching that is relatively elegant to read and implement in Ruby due to the use of the `||=` (or equals) operator.
```

Comment:
I would take out showing the implementation here. Save it for the big reveal! This is the intro after all. :-)